source/getting-started/connection.txt

```diff
@@ -0,0 +1,622 @@
+================
+Connection Guide
+================
+
+.. default-domain:: mongodb
+
+Overview
+--------
+
+This guide shows you how to connect to MongoDB with the Node.js driver using
+different connection mechanisms as well as how to configure them.
+
+When you connect to a MongoDB server, you provide your credentials and
+specific connection mechanism to the driver. The driver negotiates with the
+server to verify your identity, and the server grants you permissions
+appropriate to your user.
+
+For instructions on how to set up a new user or user permissions on your
+MongoDB server, see the **Authentication** and **Authorization** sections in
+the `Security <https://docs.mongodb.com/manual/security/>`_ server manual.
+
+The Node.js driver can connect to MongoDB using any of the `authentication
+mechanisms
+<https://docs.mongodb.com/manual/core/authentication/#authentication-mechanisms>`_
+supported by either MongoDB Community or Enterprise edition which includes:
+
+- ``DEFAULT``
+- ``SCRAM-SHA-256``
+- ``SCRAM-SHA-1``
+- ``MONGODB-CR``
+- ``X.509``
+- ``Kerberos (GSSAPI/SSPI)`` *(MongoDB Enterprise only)*
+- ``LDAP (PLAIN)`` *(MongoDB Enterprise only)*
+
+Connection Mechanisms
+---------------------
+
+In this section, you can find descriptions and sample code for connecting to
+a MongoDB server using each of the connection mechanism settings.
+
+DEFAULT
+~~~~~~~
+
+Specify the ``DEFAULT`` option to the driver in order to authenticate with the
+server using a username and password. This option is a fallback setting that
+instructs the driver to negotiate the first authentication mechanism supported
+by the server using the following order of preference:
+
+#. ``SCRAM-SHA-256``
+#. ``SCRAM-SHA-1``
+#. ``MONGODB-CR``
+
+If the ``DEFAULT`` option is specified, the driver first attempts to
+authenticate using ``SCRAM-SHA-256``. If the server does not support that
+mechanism, the driver attempts to authenticate using ``SCRAM-SHA-1``. If the
+server does not support that mechanism either, the driver attempts to
+authenticate using ``MONGODB-CR``.
+
+You can specify this option by omitting the ``authMechanism`` parameter of the
```

Comment:
I deliberated this with myself for some time. Do you think we should show the `authMechanism=DEFAULT` since it does not need to be specified? Or, does it provide potential value by explicitly including it?

Reply:
I think there's value in showing it and also in mentioning you can leave it off because it's the default. Maybe show a connection string with it, and then say "since DEFAULT is the default auth mechanism, you can also leave off the authMechanism parameter entirely and MongoDB will blabla" -- but like a better sentence than that.

I think that the current wording of "you can specify DEFAULT by omitting the parameter" is currently confusing though.